debian/marquee/nginx/conf/whatwg.conf

```diff
@@ -0,0 +1,29 @@
+ssl_session_cache shared:SSL:10m;
+ssl_session_timeout 5m;
+
+ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
+
+ssl_prefer_server_ciphers on;
+ssl_ciphers "ECDHE-RSA-AES128-GCM-SHA256 EECDH+ECDSA+AESGCM EECDH+aRSA+AESGCM EECDH+ECDSA+SHA384 EECDH+ECDSA+SHA256 EECDH+aRSA+SHA384 EECDH+aRSA+SHA256 EECDH+aRSA+RC4 EECDH EDH+aRSA RC4 !aNULL !eNULL !LOW !MD5 !EXP !PSK !SRP !DSS !MEDIUM !RC4 !CAMELLIA !3DES";
+
+ssl_dhparam /etc/ssl/certs/dhparam.pem;
+
+gzip on;
+gzip_disable "msie6";
+
+gzip_vary on;
+gzip_proxied any;
+gzip_comp_level 6;
+gzip_buffers 16 8k;
+gzip_http_version 1.1;
+gzip_types text/plain text/css application/json application/x-javascript text/xml application/xml image/svg+xml application/xml+rss text/javascript;
+
+include whatwg-headers.conf;
+
+location ~* \.(?:ico|css|js|gif|jpe?g|png|svg)$ {
```

Comment:
We do change css and js from time to time. How long would it take for them to be reloaded with this config? Or is there an etags thing involved?

Reply:
location ~* \.(?:ico|css|js|gif|jpe?g|png|svg)$ { 

> We do change css and js from time to time. How long would it take for them to be reloaded with this config?

I’m not sure. I guess I should look back and it and see

> Or is there an etags thing involved?

No, not as far as I know

Reply:
> > Or is there an etags thing involved?
>
> No, not as far as I know

So, I misspoke there. By default nginx apparently does always send an `ETag` header.

Given that, should we still send `Expires` and `Cache-Control` too?

And see https://nginx.org/en/docs/http/ngx_http_headers_module.html#expires:

> The `max` parameter sets “`Expires`” to the value “`Thu, 31 Dec 2037 23:55:55 GMT`”, and “`Cache-Control`” to 10 years.

But we are to send `Expires` and `Cache-Control` at all, it seems that as long as nginx is sending `ETag`, we might as well well max out `Expires` and `Cache-Control`.

Reply:
I’ve raised https://github.com/whatwg/misc-server/issues/10 for this what-caching-headers-to-send-for-static-stuff question

Reply:
Thanks, let's figure it out separately.